modal_proto/api.proto

```diff
@@ -1728,6 +1734,11 @@ message GeneratorDone {  // Sent as the output when a generator finishes running
   uint64 items_total = 1;
 }
 
+message GenericPayloadType {
+  ParameterType base_type = 1;
+  repeated GenericPayloadType sub_types = 2;  // sub-type for generic types like lists
```

Comment:
In a type annotation like `list[str]`, what's the "base" and what's the "sub"? "base" and "sub" are sort of synonymous so it doesn't feel super obvious.

Reply:
For a generic type `base_type` is its "origin" type (`list` in this case) in python nomenclature and the "args" are the sub types.

For a non-generic/scalar type, the `base_type` is the type itself (but notably in python `get_origin(int)` is None, so it wouldn't be an origin type, which is part of the reason I chose another name.